wavefront/server/packages/flo_cloud/flo_cloud/mssql/mssql.py

```
@@ -0,0 +1,402 @@
+            for row in data
+        ]
+        columns = list(serialized[0].keys())
+        # MSSQL has no session search_path, so an unqualified table name resolves
+        # to the connection's default schema. Qualify it with the configured schema
+        # to match the Postgres behavior of honoring self.schema.
+        if '.' not in table_name and self.schema:
+            table_name = f'{self.schema}.{table_name}'
+        quoted_table = '.'.join(f'[{part}]' for part in table_name.split('.'))
+        quoted_cols = ', '.join(f'[{col}]' for col in columns)
+        placeholders = ', '.join(f'%({col})s' for col in columns)
+        query = f'INSERT INTO {quoted_table} ({quoted_cols}) VALUES ({placeholders})'
```

Ruff (0.15.17)

[error] Possible SQL injection vector through string-based query construction

(S608)